sdk/src/main/java/co/decodable/sdk/pipeline/DecodableStreamSinkBuilder.java

```diff
@@ -9,26 +9,23 @@
 
 import co.decodable.sdk.pipeline.util.Incubating;
 import org.apache.flink.api.common.serialization.SerializationSchema;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.datastream.DataStreamSink;
 
 /** Builder for creating {@literal DecodableStreamSink} instances. */
 @Incubating
 public interface DecodableStreamSinkBuilder<T> {
 
-  /**
-   * Specifies the name of the stream to write to. Either this or {@link #withStreamId(String)} may
-   * be used, but not both.
-   */
-  DecodableStreamSinkBuilder<T> withStreamName(String streamName);
+  DecodableStreamSinkBuilder<T> withDataStream(DataStream<T> dataStream);
 
-  /**
-   * Specifies the id of the stream to write to. Either this or {@link #withStreamName(String)} may
-   * be used, but not both.
-   */
-  DecodableStreamSinkBuilder<T> withStreamId(String streamId);
+  /** Specifies the name of the stream to write to. */
+  DecodableStreamSinkBuilder<T> withStreamName(String streamName);
 
   /** Specifies the serialization schema to be used. */
   DecodableStreamSinkBuilder<T> withSerializationSchema(SerializationSchema<T> serializationSchema);
 
+  DecodableStreamSinkBuilder<T> withName(String name);
+
   /** Returns a new {@link DecodableStreamSink} for the given configuration. */
-  DecodableStreamSink<T> build();
+  DataStreamSink<T> build();
```

Comment:
`DecodableStreamSink` still is a `DataStreamSink`. The issue is that we can't name a `DataStreamSink`.

Reply:
Ok, but why is the return type changing here then?

Reply:
oh I misread. Are you saying `DecodableStreamSink` should just be a `DataStreamSink`?

Reply:
That's my question, yes. If that's what's needed to expose the name, that could be an option?

Reply:
`DataStreamSink` isn't an interface so idk how that would work

Reply:
`DecodableStreamSink` could be a class, if it has to. Taking a step back, how/where is that name used then?

Reply:
So the motivation behind all this is to be able to provide metrics (and thus, per-stream metrics) for custom pipelines. Currently, there's no way to identify, at metrics reporting time, which stream a `DecodableStreamSource` / `DecodableStreamSink` internally refers to. Also, because the SDK neither controls names or IDs, it's hard to identify the operators that actually correspond to the pipeline SDK source / sink Thus, defaulting a predictable operator name allows:

* Identify the pipeline SDK source / sink operators from the job graph
* Identify the referenced stream name and enrich metrics with stream metadata